### DISCUSSION TITLE
Ability to rollback changes

### DISCUSSION THREAD

**Discussion by Ren60FHk** [netbox-community/netbox#7613] [Ideas]
I think it would be useful to have an ability in the UI to revert some changes. For example, restore a deleted prefix or a device, revert a location's name back to previous value, etc, with a "single click".

**Comment by PieterL75** [comment]
I have been thinking the same. Some users go to a prefix ip addresses, select some ips and then click the delete button on the top right. That delete the prefix on stead of the selected IP addresses.
Easy to see what had been deleted in the logs, but a simple button on the deleted items to recreate it,. Would be very handy!

**Reply by candlerb** [reply]
I have a tendency to delete an interface, when I wanted to delete the cable from an interface :-(

**Comment by Pidpad** [comment]
I multiple times had the same issue in my company, this would be a really useful function. The amount of adrenaline pumping through the body when you realize you accidentally deleted the whole prefix and are scared that you also deleted als the IP addresses inside it is insane. Im glad that the IP's are not touched and you can simply recreate the prefix by hand and everything is ok. Don't get me wrong, we have database backups and could restore from them, but it is still scary 😄

**Reply by jeremystretch** [reply]
This thread is nearly five years old. Since then, we've introduced [NetBox branching](https://netboxlabs.com/docs/extensions/branching/) and [change management](https://netboxlabs.com/docs/developer/plugins-extensions/changes/) to serve this use case.

**Reply by Pidpad** [reply]
I will take a loot into this, thanks!
